srp_test.go

```diff
@@ -418,6 +418,38 @@ func TestCalculateU(t *testing.T) {
 	}
 }
 
+func TestSRP_MarshalBinary(t *testing.T) {
+	v := big.NewInt(42)   // value doesn't matter for this test
+	k := big.NewInt(2038) // End of the world. Value doesn't matter
+	s := NewSRPServer(KnownGroups[RFC5054Group4096], v, k)
+	data, err := s.MarshalBinary()
+	if err != nil {
+		t.Errorf("Failed to marshal SRP into binary: %+v", err)
+	}
+	if len(data) == 0 {
+		t.Errorf("Resulting byte array was empty")
+	}
+}
+
+func TestSRP_UnmarshalBinary(t *testing.T) {
+	v := big.NewInt(42)   // value doesn't matter for this test
+	k := big.NewInt(2038) // End of the world. Value doesn't matter
+	s := NewSRPServer(KnownGroups[RFC5054Group4096], v, k)
+	data, err := s.MarshalBinary()
+	if err != nil {
+		t.Errorf("Failed to marshal SRP into binary: %+v", err)
+	}
+	if len(data) == 0 {
+		t.Errorf("Resulting byte array was empty")
+	}
+
+	newServer := &SRP{}
+	err = newServer.UnmarshalBinary(data)
```

Comment:
There should definitely be an assertion to make sure the data has properly been unmarshaled and is correct.

Reply:
The test now uses the unmarshalled version of the server to perform the next steps, which in and of itself asserts that the data unmarshalled correctly otherwise the steps would not succeed.